packages/core/src/browser/progress-location-service.ts

```diff
@@ -67,13 +66,25 @@ export class ProgressLocationService implements ProgressClient {
         const show = !!progressSet.size;
         this.fireEvent(locationId, show);
     }
-    protected readonly fireEvent = throttle((locationId: string, show: boolean) => {
+
+    protected lastEvents = new Map<string, { time: number, value: boolean }>();
```

Comment:
When do we clean up `progressByLocation` and `lastEvents`?

Reply:
`progressByLocation`'s values will be cleaned at L63, but the entries remain there. 
don't see any wrong with that, as the locations can be added on demand.

regarding `lastEvents`, would you like to get the entry for `lastEvents` being wiped on done event?